src/main/kotlin/com/feedback/dao/FeedbackDao.kt

```diff
@@ -0,0 +1,75 @@
+package com.feedback.dao
+
+import com.feedback.service.FeedbackFilter
+import org.jooq.*
+import org.springframework.stereotype.Component
+
+@Component
+class FeedbackDao(private val sql: DSLContext) {
+
+    private val feedbackTable = FeedbackTable.instance
+
+    private val retrieveFeedbackTableMapper = RecordMapper<Record, RetrieveFeedbackRequest> { record ->
+        RetrieveFeedbackRequest(
+            id = record[feedbackTable.id],
+            department = record[feedbackTable.department]?.let { DepartmentType.fromString(it) },
+            comment = record[feedbackTable.comment],
+            creationTimestamp = record[feedbackTable.creationTimestamp],
+            status = record[feedbackTable.status]?.let  { StatusType.fromString(it) } ?: StatusType.NOT_SOLVED, //todo ask why can be nullable
+            employeeId = record[feedbackTable.employeeId]
+        )
+    }
+
+    fun insertFeedback(userDetails: LoggedInUser, feedback: FeedbackCreationRequest): Long {
```

Comment:
that's ok to stay with "feedback", as a variable name, unlike the one you called "employee".
👍

Reply:
I understand that is because the name say so on it own, correct?

Reply:
yes, you insert a feedback. clear enough